pyproject.toml

```diff
@@ -1,14 +1,95 @@
-# For more information about this pyproject.toml file, see
-# PEP 518: https://www.python.org/dev/peps/pep-0518/
+################################################################################
+# PEP 621 Project Metadata                                                     #
+################################################################################
+
+# see https://peps.python.org/pep-0621/
+
+[project]
+name = "probnum"
+description = "Probabilistic Numerics in Python."
+readme = "README.md"
+requires-python = ">=3.8,<3.11"
+license = {file = "LICENSE.txt"}
+authors = [
```

Comment:
I realize having specific names and e-mail addresses here is probably part of the setuptools specification, but I am not a huge fan, since it will cause PRs when maintainers change and may not stay up-to-date. (I realize similar issues exist with the docs). I somewhat prefer the option to just specify this on GitHub and let the contributors list show who creates this package.

Reply:
In this case we should create a joint mail address for all ProbNum maintainers soon, since this is the only piece of contact information listed on PyPI. It also seems to be quite common in other open source project (see NumPy, SciPy, PyTorch)